src/util/route.js

```diff
@@ -6,13 +6,13 @@ export function isSameRoute (a: Route, b: ?Route): boolean {
   } else if (a.path && b.path) {
     return (
       a.path === b.path &&
-      a.hash === b.hash &&
+      (a.hash || '') === (b.hash || '') &&
```

Comment:
@posva Tried that, but flow still doesn't give warnings : (
and even if we make it give warnings, we still have to fix passing `Location` to a function that requires `Route`..